blueflood-elasticsearch/src/test/java/com/rackspacecloud/blueflood/service/ElasticIOTest.java

```diff
@@ -138,6 +137,35 @@ public void testWildcardForPreaggregatedMetric() throws Exception {
         testWildcard(TENANT_C, null);
     }
 
+    @Test
+    public void testBatchQueryWithNoWildCards() throws Exception {
+        String tenantId = TENANT_A;
+        String query1 = "one.two.three00.fourA.five1";
+        String query2 = "one.two.three01.fourA.five2";
+        List<SearchResult> results;
+        ArrayList<String> queries = new ArrayList<String>();
+        queries.add(query1);
+        queries.add(query2);
+        results = elasticIO.search(tenantId, queries);
+        Assert.assertEquals(results.size(), 2); //we searched for 2 unique metrics
+        results.contains(new SearchResult(TENANT_A, query1, UNIT));
+        results.contains(new SearchResult(TENANT_A, query2, UNIT));
+    }
+
```

Comment:
@tilogaat If the query2 = "foo.bar" i.e we are querying for something which does not exist, then it will match query1 and return the results of just that query.